src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -233,7 +233,7 @@ Mac::TxFrame *CslTxScheduler::HandleFrameRequest(Mac::TxFrames &aTxFrames)
 
     VerifyOrExit(delay <= mCslFrameRequestAheadUs + kFramePreparationGuardInterval, frame = nullptr);
 
-    frame->SetTxDelay(txDelay);
+    frame->SetTxDelay(txDelay - kRadioHeaderDuration);
```

Comment:
This looks problematic; whether `txDelay` is always greater than 192 or not depends I think on a config value (`OPENTHREAD_CONFIG_MAC_CSL_REQUEST_AHEAD_US`). Maybe a static assert can be added that validates `OPENTHREAD_CONFIG_MAC_CSL_REQUEST_AHEAD_US >= kRadioHeaderDuration` during compile time?

Reply:
Oops, my comment looks wrong now. It was based on the wrong assumption that `txDelay >= OPENTHREAD_CONFIG_MAC_CSL_REQUEST_AHEAD_US` due to the above line `VerifyOrExit`, but the test is not for `<=` and not for `>=`. 

Looking at how `txDelay` is calculated, it's very likely that `txDelay >= kRadioHeaderDuration` holds.  But: are we 100% sure about this?  Ideally in case of `txDelay < kRadioHeaderDuration` we should just delay an amount of `0` which is fine because the CSL Transmitter will be a bit late in sending the frame but the CSL Receiver can receive it nevertheless - it has some window it keeps waiting for it.

So something like:
`frame->SetTxDelay(  (txDelay > kRadioHeaderDuration) ? (txDelay-kRadioHeaderDuration) : 0  );`

Reply:
Agree, done.